src/imap.rs

```diff
@@ -114,6 +115,24 @@ impl Default for ImapConfig {
     }
 }
 
+#[derive(Debug, Fail, Clone, Eq, PartialEq)]
+enum SelectError {
+    #[fail(display = "Could not obtain imap-session object.")]
+    NoSession,
+
+    #[fail(display = "Connection Lost or no connection established")]
+    ConnectionLost,
+
+    #[fail(display = "imap-close (to expunge messages) failed: {}", _0)]
+    CloseExpungeFailed(String),
+
+    #[fail(display = "Folder name invalid: {:?}", _0)]
+    BadFolderName(String),
```

Comment:
Convert this into a struct so that the `String` has a descriptive field name, like `folder`.

Reply:
Doesn't the `BadFolderName` already tell that it contains a folder name?

Reply:
True, it is only hard to understand because there is a `CloseExpungeFailed` that has no cause annotation and also contains a String. Once it is fixed, the purpose of the String inside a BadFolderName is clear.